tools/ARIAtools/ARIAProduct.py

```diff
@@ -221,6 +222,16 @@ def __continuous_time__(self):
         if len(self.products)==1:
             sorted_products.extend([[dict(zip(self.products[0][0].keys(), [list(a) for a in zip(self.products[0][0].values())])), dict(zip(self.products[0][1].keys(), [list(a) for a in zip(self.products[0][1].values())]))]])
 
+        # Check for (and remove) duplicate products
```

Comment:
@sssangha Do you have an idea how often this is occurring.
We have not done a thorough test, and want to make sure of this is stable for back-porting.
It could be considered enhancement for the future rather than a bugfix.

Reply:
There seem to be ~2-5 cases early on in the mission phase (e.g. 2015-early 2016) from the few tracks I've looked at. 

This approach should be stable though, I even made sure to track for duplicate products in an independent for loop above the one which the final, updated dictionary is passed.

Reply:
Also, should I perhaps provide you guys examples of specific duplicate products for testing?

Reply:
if you could present a few examples that might be good.

Reply:
Yup, here are a couple of download links:
wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2015/03/01/S1-GUNW-A-R-041-tops-20150301_20141231-115811-32793N_30910N-PP-1a3a-v2_0_1/S1-GUNW-A-R-041-tops-20150301_20141231-115811-32793N_30910N-PP-1a3a-v2_0_1.nc
wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2015/03/01/S1-GUNW-A-R-041-tops-20150301_20141231-115811-32793N_30910N-PP-3ba0-v2_0_1/S1-GUNW-A-R-041-tops-20150301_20141231-115811-32793N_30910N-PP-3ba0-v2_0_1.nc

wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2015/03/01/S1-GUNW-A-R-041-tops-20150301_20141231-115836-34285N_32405N-PP-6aa3-v2_0_1/S1-GUNW-A-R-041-tops-20150301_20141231-115836-34285N_32405N-PP-6aa3-v2_0_1.nc
wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2015/03/01/S1-GUNW-A-R-041-tops-20150301_20141231-115836-34285N_32405N-PP-b7ba-v2_0_1/S1-GUNW-A-R-041-tops-20150301_20141231-115836-34285N_32405N-PP-b7ba-v2_0_1.nc

Reply:
Ok @dbekaert, as we discussed earlier today I found a fringe case on track 77 where we have duplicate products over a track with sparse land coverage:
wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2017/12/17/S1-GUNW-A-R-077-tops-20171217_20171205-231330-33719N_31670N-PP-b8ec-v2_0_1/S1-GUNW-A-R-077-tops-20171217_20171205-231330-33719N_31670N-PP-b8ec-v2_0_1.nc

wget --no-check-certificate --mirror -np -nH --reject "index.html*" --cut-dirs=7 http://hysds-aria.s3-website-us-west-2.amazonaws.com/datasets/interferogram/v2.0.1/2017/12/17/S1-GUNW-A-R-077-tops-20171217_20171205-231330-33719N_31670N-PP-bc89-v2_0_1/S1-GUNW-A-R-077-tops-20171217_20171205-231330-33719N_31670N-PP-bc89-v2_0_1.nc


The code is able to handle such a case appropriately. However, I found a small bug in the current logic of the code. There is a check in the ARIAProduct class to see if there is only one product in the list, to avoid entering the "for" loop comparing the temporal connectively of successive products. This check must be moved after the section where duplicate products are removed, otherwise the sorted dictionary will be empty.

Once this bug-fix is accepted, I will backport it to dev.